agent/pkg/proto/agent.proto

```diff
@@ -0,0 +1,272 @@
+syntax = "proto3";
+
+import "google/protobuf/struct.proto";
+import "google/protobuf/timestamp.proto";
+
+option go_package = "github.com/pulumi/pulumi-kubernetes-operator/agent/pkg/proto";
+
+package agent;
+
+service AutomationService {
+
+  /**
+   * WhoAmI returns detailed information about the currently logged-in Pulumi
+   * identity.
+   */
+  rpc WhoAmI(WhoAmIRequest) returns (WhoAmIResult) {}
+
+  /**
+   * Info returns information about the given stack.
+   */
+  rpc SelectStack(SelectStackRequest) returns (SelectStackResult) {}
+
+  /**
+   * Info returns information about the given stack.
+   */
+  rpc Info(InfoRequest) returns (InfoResult) {}
+
+  /**
+   * SetAllConfig sets multiple configuration values for a stack.
+   */
+  rpc SetAllConfig(SetAllConfigRequest) returns (SetAllConfigResult) {}
+
+  /**
+   * Install installs the Pulumi plugins and dependencies.
+   */
+  rpc Install(InstallRequest) returns (InstallResult) {}
+
+  /**
+   * Preview performs a dry-run update to a stack, returning the expected
+   * changes.
+   */
+  rpc Preview(PreviewRequest) returns (stream PreviewStream) {}
+
+  /**
+   * Refresh updates the resources in a stack to match the current state of the
+   * cloud provider.
+   */
+  rpc Refresh(RefreshRequest) returns (stream RefreshStream) {}
+
+  /**
+   * Up creates or updates the resources in a stack, returning the actual
+   * changes.
+   */
+  rpc Up(UpRequest) returns (stream UpStream) {}
+
+  /**
+   * Destroy deletes all resources in a stack.
+   */
+  rpc Destroy(DestroyRequest) returns (stream DestroyStream) {}
```

Comment:
I think the stream types can also be called responses since the streaming-ness is obvious from the context.